[bug] Args passed to runApp are ignored by Typesafe Config when another config was previously loaded

**Description**

Since Jooby 4.5:  if we are using typesafe to read some configuration file before starting the jooby application, arguments passed to Jooby.runApp will no longer be visible during configuration resolution. 

Jooby will internally set these args as system properties but, since Typesafe Config has already loaded and cached system properties before runApp initializes the application, it causes the args passed to runApp to be effectively ignored.

This started happening due to recent changes to Environment.systemProperties(), which will no longer pass the System.getProperties() as an argument.

**To Reproduce**

Java file:
```
public class TestIssue extends Jooby {

    {
        getConfig();
    }

    public static void main(String[] args) {

        // loads some auxiliary configuration file --> "freezes" the typesafe cache
        ConfigFactory.load("variables.conf");

        // used to replace a placeholder on the config file
        String[] newArgs = {
                "bd.url=localhost"
        };

        // internally will load the application.conf properties
        runApp(newArgs, TestIssue::new);
    }
}
```

application.conf:
```
persistence.url = ${bd.url}
```

variables.conf:
```
auth_client.secret = "xxx" 
```

**Expected behavior**
Arguments passed to runApp should be available as system properties, allowing for the replacement of the placeholder ${bd.url}, just as they were in previous versions.

**Actual behavior**
The substituion of ${bd.url} never occurs and the application throws an error on boot:

Caused by: com.typesafe.config.ConfigException$UnresolvedSubstitution: application.conf @ file:[...]/application.conf: 1: Could not resolve substitution to a value: ${bd.url}


**Environment**
Jooby version: 4.5.0
Java version: 21.0.10
OS: Ubuntu 24.04.4

--- a/jooby/src/main/java/io/jooby/Environment.java
+++ b/jooby/src/main/java/io/jooby/Environment.java
@@ -17,6 +17,7 @@ import org.jspecify.annotations.Nullable;
 import com.typesafe.config.Config;
 import com.typesafe.config.ConfigException;
 import com.typesafe.config.ConfigFactory;
+import com.typesafe.config.ConfigParseOptions;
 
 /**
  * Application environment contains configuration object and active environment names.
@@ -239,7 +240,17 @@ public class Environment {
    * @return Configuration object.
    */
   public static Config systemProperties() {
-    return ConfigFactory.systemProperties();
+    var systemPropertiesCopy = new Properties();
+    for (var entry : System.getProperties().entrySet()) {
+      // Java 11 introduces 'java.version.date', but we don't want that to
+      // overwrite 'java.version'
+      if (!entry.getKey().toString().startsWith("java.version.")) {
+        systemPropertiesCopy.put(entry.getKey(), entry.getValue());
+      }
+    }
+    return ConfigFactory.parseProperties(
+        systemPropertiesCopy,
+        ConfigParseOptions.defaults().setOriginDescription("system properties"));
   }
 
   /**

--- a/jooby/src/test/java/io/jooby/EnvironmentTest.java
+++ b/jooby/src/test/java/io/jooby/EnvironmentTest.java
@@ -276,7 +276,6 @@ public class EnvironmentTest {
   @Test
   public void testSystemPropertiesAndEnv() {
     Config props = Environment.systemProperties();
-    System.out.println(props.getAnyRef("java.version"));
     assertEquals(System.getProperty("java.version"), props.getString("java.version"));
 
     Config env = Environment.systemEnv();

--- a/jooby/src/test/java/io/jooby/Issue3950.java
+++ b/jooby/src/test/java/io/jooby/Issue3950.java
@@ -1,0 +1,36 @@
+/*
+ * Jooby https://jooby.io
+ * Apache License Version 2.0 https://jooby.io/LICENSE.txt
+ * Copyright 2014 Edgar Espina
+ */
+package io.jooby;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+
+import com.typesafe.config.ConfigFactory;
+
+public class Issue3950 {
+
+  @Test
+  void shouldLoadSystemPropertiesAfterFreeze() {
+    // Not here before
+    assertFalse(ConfigFactory.systemProperties().hasPath("auth_client_secret"));
+    ConfigFactory.load(ConfigFactory.parseMap(Map.of("bd", Map.of("url", "localhost"))));
+    // Not there after
+    assertFalse(ConfigFactory.systemProperties().hasPath("auth_client_secret"));
+
+    System.setProperty("auth_client_secret", "xxx");
+
+    var env = Environment.loadEnvironment(new EnvironmentOptions());
+    var config = env.getConfig();
+    // Sync from environment
+    assertEquals("xxx", config.getString("auth_client_secret"));
+    // But still not there on config system properties
+    assertFalse(ConfigFactory.systemProperties().hasPath("auth_client_secret"));
+  }
+}

